pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -585,6 +585,16 @@ type TiKVSpec struct {
 	// EnableNamedStatusPort enables status port(20180) in the Pod spec.
 	// If you set it to `true` for an existing cluster, the TiKV cluster will be rolling updated.
 	EnableNamedStatusPort bool `json:"enableNamedStatusPort,omitempty"`
+
+	// ScaleInParallelism configures max scale in replicas for TiKV stores.
+	// +kubebuilder:default=1
+	// +optional
+	ScaleInParallelism *int32 `json:"scaleInParallelism,omitempty"`
+
+	// ScaleOutParallelism configures max scale out replicas for TiKV stores.
+	// +kubebuilder:default=1
+	// +optional
+	ScaleOutParallelism *int32 `json:"scaleOutParallelism,omitempty"`
```

Comment:
How about combining `ScaleInParallelism` and `ScaleOutParallelism` into a field called `ScalePolicy`?

Reply:
For the very long ago discussion we use these two variables, I think both ways are acceptable, what do you think? @DanielZhangQD

Reply:
@lizhemingi What @liubog2008 means is to define a struct as below:
```
type ScalePolicy struct {
  ScaleOutParallelism
  ScaleInParallelism
}
```
Just in case we may add more fields for scaling later, I think this is reasonable, WDYT?
Sorry that this comment is raised so late.

Reply:
Yeah, I get the point of @liubog2008. I think both the new and old one are ok. 

I will do it later today.

Reply:
I have changed the implementation to `ScalePolicy`.

But the changes of `hack/update-all.sh` seems too much, could you please take a look and let me know how to fix it if needed.

Reply:
> the changes of hack/update-all.sh seems too much

You mean there are some problems when the script `hack/update-all.sh` is running? I'm sorry I can't understand what `too much changes` means.

Reply:
I cannot find unexpected changes in your PR.